### ISSUE TITLE
setting x_scale symlog distorts graph

### ISSUE THREAD

**Issue by pjkrauspe** [matplotlib/matplotlib#14597]
### Bug report

EDIT:
I figured out it's due to the standard set as the option 'linscalex'.
Good addition would be force entry of linscalex when using symlog or throw a flag that linscalex is applied.
/EDIT

setting xscale to symlog produces a different graph than setting it to log

```python

def func(a):
    return a**2*np.exp(a)
alpha=np.linspace(1,10,1000)
plt.plot(alpha,1e5*func(a=alpha))
axes=plt.gca()
axes.set_yscale('log')
axes.set_xscale('symlog')
plt.xlim(1,10)
plt.title('symlog')
plt.show()

plt.plot(alpha,1e5*func(a=alpha))
axes=plt.gca()
axes.set_yscale('log')
axes.set_xscale('log')
plt.xlim(1,10)
plt.title('log')

plt.show()


```

**Actual outcome**
![symlog](https://user-images.githubusercontent.com/52042769/59918214-55291500-9424-11e9-9f31-2c5e47494655.png)

![log](https://user-images.githubusercontent.com/52042769/59918235-63773100-9424-11e9-86c6-cc6912d8a207.png)

**Expected outcome**

I expect this to show what I see when I plot 'log'. 

**Matplotlib version**

  * Operating system: MAC OS X, reproduced on Ubuntu 16.04, too
  * Matplotlib version: 3.1.0
  * Python version: 3.7
  * Jupyter version (if applicable): 5.6.8

**Comment by timhoffm** [comment]
I think explicitly requiring *linthresx* is the right way to. I don't think there is a reasonable default.
One could try to infer a value from the already plotted data, but that would change the plot depending on when the log scale is set, and I'd rather not go down that road.

**Comment by timhoffm** [comment]
Related, but discarded: Adding a warning #14297

**Comment by timhoffm** [comment]
Could be more easily addressed after #14916.

**Comment by jklymak** [comment]
I agree that there is no sensible default.  This really should never have been a string-only option to `set_x/yscale`.  I'm going to close this though, because googling "symlog scale matplotlib" brings up https://matplotlib.org/gallery/scales/symlog_demo.html#sphx-glr-gallery-scales-symlog-demo-py which is very clear about the use of `linthresh`
